### PR TITLE
drivers: i3c: npcx: add HDR-DDR mode for transfer

### PR DESCRIPTION
**Pull request by alvsun**
1. Support HDR-DDR transfer.
2. Update IBI error handling.

### CODE REVIEW

File: drivers/i3c/i3c_npcx.c [hunk 1]
```diff
@@ -396,6 +400,56 @@ static inline int npcx_i3c_request_emit_stop(struct i3c_reg *inst)
...
+	return 0;
+}
+
+static inline int npcx_i3c_xfer_stop(struct i3c_reg *inst)
+{
+	uint32_t state;
+	int ret;
+
+	state = npcx_i3c_state_get(inst);
+	LOG_DBG("%s, state=%d", __func__, state);
```
Comment (fabiobaltieri) [inline drivers/i3c/i3c_npcx.c:433]: do you need __func_ with LOG_DBG? Thought it was added automatically

**Review by fabiobaltieri — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

File: drivers/i3c/i3c_npcx.c (hunk 1 above)
Comment (alvsun) [reply drivers/i3c/i3c_npcx.c]: Done

File: drivers/i3c/i3c_npcx.c [hunk 2]
```diff
@@ -958,35 +1043,63 @@ static int npcx_i3c_transfer(const struct device *dev, struct i3c_device_desc *t
...
+					"len=%d", msgs[i].num_xfer);
+				ret = -EINVAL;
 				break;
 			}
-			send_broadcast = false;
+		} else {
+			LOG_ERR("%s: Controller HDR Mode %d", dev->name,
+				data->common.ctrl_config.supported_hdr);
+			LOG_ERR("%s: msg HDR Mode %#x, msg flag %#x", __func__, msgs[i].hdr_mode,
+				msgs[i].flags);
```
Comment (XenuIsWatching) [inline drivers/i3c/i3c_npcx.c:1089]: 2 error logs here??, also one looks like it's using dev->name before the `:` while the other is using `__func__`

**Review by XenuIsWatching — CHANGES_REQUESTED** [review]
looks good, just a few minor issues here and there

File: include/zephyr/drivers/i3c.h [hunk 3]
```diff
@@ -498,6 +498,16 @@ struct i3c_msg {
 	 * Use SDR mode if none is set.
 	 */
 	uint8_t			hdr_mode;
+
+	/**
+	 * HDR command code field (7-bit)
+	 *
+	 * For DDR,
```
[the remark this answers is not in the document]
Comment (XenuIsWatching) [reply include/zephyr/drivers/i3c.h:505]: also... should the helper inline function be `0`'ing these?
https://github.com/zephyrproject-rtos/zephyr/blob/c211cb347e0af0a4931e0e7af3d93577bcc7af8f/include/zephyr/drivers/i3c.h#L1825

[the remark this answers is not in the document]
Comment (alvsun) [reply include/zephyr/drivers/i3c.h]: Done

File: include/zephyr/drivers/i3c.h [hunk 4]
```diff
@@ -498,6 +498,16 @@ struct i3c_msg {
...
+	 * Write: 8’h00 to 8’h7F (MSB 0 for write)
+	 * Read : 8’h80 to 8’hFF (MSB 1 for read)
+	 *
+	 */
+	uint8_t			hdr_cmd_code;
```
[the remark this answers is not in the document]
Comment (alvsun) [reply include/zephyr/drivers/i3c.h]: Done

File: drivers/i3c/i3c_npcx.c (hunk 2 above)
Comment (alvsun) [reply drivers/i3c/i3c_npcx.c]: Done

**Review by XenuIsWatching — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**